lib/cfn-nag/custom_rules/password_base_rule.rb

```diff
@@ -31,32 +31,34 @@ def audit_impl(cfn_model)
 
     violating_resources.map(&:logical_resource_id)
   end
-end
 
-private
+  private
 
-def verify_parameter_exists(resource, password_property, sub_property_name)
-  if sub_property_name.nil?
-    resource.send(password_property).nil?
-  else
-    resource.send(password_property)[sub_property_name].nil?
+  def property_does_not_exist(resource, password_property, sub_property_name)
+    if resource.send(password_property).nil?
+      true
+    elsif sub_property_name.nil?
+      resource.send(password_property).nil?
```

Comment:
if we have fallen through to here, don't we know that password_property == nil?  in combination isn't the result true if password_property or sub_property == nil?

Reply:
If we make it to line 40 here then it means that the `password_property` does exist, but if the `sub_property_name` == nil? then we can return `false` since the property **would** exist.

I'll push a change so that we don't have to check `resource.send(password_property).nil?` twice here.